File: ruby/sieve.rb
class Sieve

  attr_reader :upper_bound

  def initialize(upper_bound)
    @upper_bound=upper_bound
  end

  def primes(divisor=2, group=(2..upper_bound).to_a)
    return [] if upper_bound < 2
    group.delete_if{|num| num%divisor==0 && num!=divisor}
    divisor == 2 ? divisor+=1 : divisor+=2
    divisor +=2 until group.include?(divisor) || divisor>(upper_bound/2)
    divisor>(upper_bound/2) ? group : primes(divisor, group)
  end


end

File: js/sieve.js
Sieve = function(num){
  this.num = num
  var eligible = [...Array(this.num+1).keys()].slice(2, this.num+1)
  this.primes=findPrimes(eligible)
};

function findPrimes(group, divisor=2){
  if (group[group.length-1]/2 <= divisor){return group};
  var group = group.filter(function(value){
    return value%divisor !== 0 || value === divisor
  });
  divisor === 2 ? divisor+=1 : divisor+=2
  return findPrimes(group, divisor);
};

module.exports = Sieve;
